src/main/java/org/opentripplanner/middleware/utils/ItineraryExistenceChecker.java

```diff
@@ -0,0 +1,66 @@
+package org.opentripplanner.middleware.utils;
+
+import org.opentripplanner.middleware.otp.OtpDispatcher;
+import org.opentripplanner.middleware.otp.OtpDispatcherResponse;
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.OtpResponse;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * This utility class checks for existence of itineraries for given OTP queries.
+ */
+public class ItineraryExistenceChecker {
+    private final Function<String, OtpDispatcherResponse> otpDispatcherFunction;
+
+    /**
+     * @param otpDispatcherFunction a function that takes String as input and returns {@link OtpDispatcherResponse},
+     *                              example: {@link OtpDispatcher#sendOtpPlanRequest}.
+     */
+    public ItineraryExistenceChecker(Function<String, OtpDispatcherResponse> otpDispatcherFunction) {
+        if (otpDispatcherFunction == null) throw new NullPointerException();
+        this.otpDispatcherFunction = otpDispatcherFunction;
+    }
```

Comment:
I'm not a huge fan of this constructor because it appears the sole purpose of it is to inject testing code. I think it'd be better to have the tests always use the `OtpDispatcher` and then properly mock some requests/responses using the `TestUtils.setupOtpMocks` method.

Reply:
Using mpckOtpServer as of https://github.com/ibi-group/otp-middleware/pull/77/commits/81d7fa65c0868e2811e6aa4889919b9e419ecb71